src/Storages/MergeTree/MergeTreeSelectProcessor.cpp

```diff
@@ -245,7 +251,7 @@ ChunkAndProgress MergeTreeSelectProcessor::read()
             if (add_part_level)
                 chunk.getChunkInfos().add(std::make_shared<MergeTreeReadInfo>(data_part->info.level));
 
-            if (reader_settings.use_query_condition_cache)
+            if (query_condition_cache_writer)
```

Comment:
Is it okay that we do not enter this branch w/o `prewhere_info`? This is a bit suspicious because previously we would enter here.